### ISSUE TITLE
ERROR  Unable to find any random port on host

### ISSUE THREAD

**Issue by stefanobartoletti** [nuxt/cli#337]
### Environment

```
> npx nuxi info
Working directory: /home/stefano/Sviluppo/Personali/nuxt-app                                                                                20:50:53
Nuxt project info:                                                                                                                          20:50:53

------------------------------
- Operating System: Linux
- Node Version:     v18.19.0
- Nuxt Version:     3.10.1
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  pnpm@8.15.1
- Builder:          -
- User Config:      devtools
- Runtime Modules:  -
- Build Modules:    -
------------------------------

👉 Report an issue: https://github.com/nuxt/nuxt/issues/new                                                                                 20:50:53

👉 Suggest an improvement: https://github.com/nuxt/nuxt/discussions/new

👉 Read documentation: https://nuxt.com

```

### Reproduction

It happens with every Nuxt project.

In this example I've only created a new one with `npx nuxi init`

### Describe the bug

Nuxt is unable to start and open any development port on my current configuration.
I've recently reinstalled my system, and before this everything was working as expected. 

Some more context:

- It only happens with Nuxt, other apps can open port `3000` (i.e. `npx serve .` works correctly with the same exact port)
- It happens with firewall both turned on and off (but it does not seem to be relevant)
- Port `3000` is not already used by any other app of service (see below)

**Error**
```
>$> pnpm dev

> nuxt-app@ dev /home/stefano/Sviluppo/Personali/nuxt-app
> nuxt dev

Nuxt 3.10.1 with Nitro 2.8.1                                                                                                                20:48:43
[get-port] Unable to find an available port (tried 3000-3100 on host "slimbook-laptop"). Using alternative port undefined                   20:48:46

 ERROR  Unable to find any random port on host "slimbook-laptop"                                                                            20:48:46

  at getRandomPort (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/chunks/index2.mjs:343:11)
  at async getPort (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/chunks/index2.mjs:321:21)
  at async listen (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/chunks/index2.mjs:29727:38)
  at async _createDevProxy (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/chunks/dev.mjs:229:20)
  at async Object.run (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/chunks/dev.mjs:182:24)
  at async runCommand$1 (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/shared/nuxi.4fde776c.mjs:1648:16)
  at async runCommand$1 (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/shared/nuxi.4fde776c.mjs:1639:11)
  at async runMain$1 (node_modules/.pnpm/nuxi@3.10.0/node_modules/nuxi/dist/shared/nuxi.4fde776c.mjs:1773:7) 



 ERROR  Unable to find any random port on host "slimbook-laptop"                                                                            20:48:46

 ELIFECYCLE  Command failed with exit code 1.

```

**Used Ports**
```
> lsof -i -P | grep -i "listen"
kdeconnec  2477 stefano  17u  IPv6  16775      0t0  TCP *:1716 (LISTEN)
dropbox   31617 stefano  50u  IPv6  70874      0t0  TCP *:17500 (LISTEN)
dropbox   31617 stefano  51u  IPv4  70875      0t0  TCP *:17500 (LISTEN)
```

I'd like some suggestions to fix this. Let me know if you need more info or context. Like I said, it happens also with a fresh Nuxt installation, without changing a single line from what `nuxi` gives me.

**Comment by stefanobartoletti** [comment]
I did a little more research, and looking at past closed issues, #19068 and #20019 seem to be related (or at least they look similar)

I tried also running 

`PORT=3000 pnpm dev` 
`pnpm dev --port 3000`
`NUXT_HOST="127.0.0.1" pnpm dev`
`NUXT_HOST="" pnpm dev`

but the problems is still there. It looks like Nuxt is unable to perform the action of opening a port, rather than a problem with a specific port number itself.

**Comment by stefanobartoletti** [comment]
Sorry to keep spamming, but I'm doing some more tests.

Running `pnpm generate` and `pnpm preview` work correctly, the site is served without issues:

```
> pnpm preview

> nuxt-app@ preview /home/stefano/Sviluppo/Personali/nuxt-app
> nuxt preview

                                                                                                                                            21:41:14
 ╭─────────────────────────────────────Preview Mode────────────────────────────────────────╮
 │                                                                                         │
 │  You are running Nuxt production build in preview mode.                                 │
 │  For production deployments, please directly use npx serve ./public command.            │
 │                                                                                         │
 │  Node.js:           v18.19.0                                                            │
 │  Nitro Preset:      static                                                              │
 │  Working directory: .output                                                             │
 │                                                                                         │
 ╰─────────────────────────────────────────────────────────────────────────────────────────╯

ℹ Loading .env. This will not be loaded when running the server in production.                                                             21:41:14
ℹ Starting preview command: npx serve ./public                                                                                             21:41:14
                                                                                                                                            21:41:14
 INFO  Accepting connections at http://localhost:3000
 HTTP  12/2/2024 21:41:20 127.0.0.1 GET /
 HTTP  12/2/2024 21:41:20 127.0.0.1 Returned 200 in 11 ms
 HTTP  12/2/2024 21:41:20 127.0.0.1 GET /_nuxt/entry.CW_rqhLt.css
 HTTP  12/2/2024 21:41:20 127.0.0.1 GET /_nuxt/entry.BLToI8N4.js
 HTTP  12/2/2024 21:41:20 127.0.0.1 GET /_payload.json
 HTTP  12/2/2024 21:41:20 127.0.0.1 Returned 200 in 2 ms
 HTTP  12/2/2024 21:41:20 127.0.0.1 Returned 200 in 13 ms
 HTTP  12/2/2024 21:41:20 127.0.0.1 Returned 200 in 13 ms
 HTTP  12/2/2024 21:41:21 127.0.0.1 GET /favicon.ico
 HTTP  12/2/2024 21:41:21 127.0.0.1 Returned 200 in 2 ms
 HTTP  12/2/2024 21:41:21 127.0.0.1 GET /_nuxt/builds/meta/a3e8ed04-4a7e-4d05-8e57-de7690eaa898.json
 HTTP  12/2/2024 21:41:21 127.0.0.1 Returned 200 in 1 ms
```

The problem lies specifically with the dev server

**Comment by stefanobartoletti** [comment]
Ok very silly me, for some reason my system uses the hostname (i.e. `slimbook-laptop`) as the primary address instead of `localhost`.

I had to edit my `/etc/hosts` and add this line

```
127.0.0.1   slimbook-laptop
```

now the names are resolved correctly and I can run the dev server correctly.

If it is not a problem I would keep this issue ope for a couple of days and see if everything goes smoothly, then close it after a bit of real life testing (but it should be solved 99% I think)

**Comment by Gregor-prog** [comment]
@stefanobartoletti  thanks for the guide
to give a structured approach, you'd first have to install pnpm globally "npm i pnpm -g".
then run "pnpm generate"
then "pnpm review"
then "pnpm dev --host 127 .0.0.1"

**Comment by stefanobartoletti** [comment]
Is is not strict necessary to install on globally if using corepack

**Comment by imzedi** [comment]
For me,

pnpm dev --host 127.0.0.1 --port 80

```
Nuxt 3.17.7 with Nitro 2.12.0                                                                                                                                  nuxi 1:27:42 PM
[get-port] Unable to find an available port (tried 80 on host "127.0.0.1"). Using alternative port 3000.                                                            1:27:42 PM
                                                                                                                                                                    1:27:42 PM
  ➜ Local:    http://127.0.0.1:3000/
  ➜ Network:  use --host to expose
```

pnpm dev --host 0.0.0.0 --port 80  

```
Nuxt 3.17.7 with Nitro 2.12.0  
  ➜ Local:    http://0.0.0.0:/
  ➜ Network:  http://169.254.73.89:/ [QR code]
  ➜ Network:  http://172.20.10.5:/
```

Not working on port 80 on localhost; 

- I don't have any NUXT_HOST env
- I have commented out devServer key in config as thats not been picked up
   ```
  // devServer: {
  //   port: 80,
  //   host: '0.0.0.0'
  // },
   ```

- Previously it was working with v3.17.4
- After upgrading to v3.17.7 its not working

**Comment by imzedi** [comment]
> > For me,
> > pnpm dev --host 127.0.0.1 --port 80
> > ```
> > Nuxt 3.17.7 with Nitro 2.12.0                                                                                                                                  nuxi 1:27:42 PM
> > [get-port] Unable to find an available port (tried 80 on host "127.0.0.1"). Using alternative port 3000.                                                            1:27:42 PM
> >                                                                                                                                                                     1:27:42 PM
> >   ➜ Local:    http://127.0.0.1:3000/
> >   ➜ Network:  use --host to expose
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > pnpm dev --host 0.0.0.0 --port 80
> > ```
> > Nuxt 3.17.7 with Nitro 2.12.0  
> >   ➜ Local:    http://0.0.0.0:/
> >   ➜ Network:  http://169.254.73.89:/ [QR code]
> >   ➜ Network:  http://172.20.10.5:/
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Not working on port 80 on localhost;
> > 
> > * I don't have any NUXT_HOST env
> > * I have commented out devServer key in config as thats not been picked up
> >   ```
> >   // devServer: {
> >   //   port: 80,
> >   //   host: '0.0.0.0'
> >   // },
> >   ```
> > * Previously it was working with v3.17.4
> > * After upgrading to v3.17.7 its not working
> 
> Thing is it was working with exposed network; with that version too, I cannot run on localhost with port 80

So problem was this
On UNIX-like systems, ports below 1024 (like 80) are considered privileged, it should only be run by root user or with permission to current user.

Confirmed with running it using sudo and it did worked
anyways I will use network exposed as I mostly test on other devices.
